src/shared/sam/debugger/awsSamDebugConfiguration.ts

```diff
@@ -130,14 +134,13 @@ export function createTemplateAwsSamDebugConfig(
             lambda:
                 preloadedConfig.environmentVariables || preloadedConfig.eventJson
                     ? {
-                          event: preloadedConfig.eventJson
-                              ? {
-                                    json: preloadedConfig.eventJson,
-                                }
-                              : undefined,
+                          event: preloadedConfig.eventJson ? { json: preloadedConfig.eventJson } : {},
                           environmentVariables: preloadedConfig.environmentVariables,
                       }
-                    : undefined,
+                    : {
```

Comment:
Seems unnecessary to have nested if statements considering the top one already handles the base case of no properties. Could you get rid of the outer one and have something like:
```
{
event: preloadedConfig.hasOwnProperty('eventJson') ? { json: preloadedConfig.eventJson } : {},
environmentVariables: preloadedConfig.hasOwnProperty('environmentVariables') ? preloadedConfig.environmentVariables : {},
}
```